File: devops/experimental/javascript/src/hello.js
export const HELLO = "hello world";

export function extractWorld(greeting) {
  if (typeof greeting !== "string") {
    return null;
  }
  const needle = "world";
  const idx = greeting.indexOf(needle);
  if (idx === -1) {
    return null;
  }
  return greeting.slice(idx, idx + needle.length);
}

File: devops/experimental/golang/hello.go
package hello

import "strings"

const Hello = "hello world"

func ExtractWorld(greeting string) (string, bool) {
	idx := strings.Index(greeting, "world")
	if idx == -1 {
		return "", false
	}

	word := greeting[idx : idx+len("world")]
	return word, true
}
